hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/profiling/DefaultConsumerProfiler.java

```diff
@@ -0,0 +1,79 @@
+package pl.allegro.tech.hermes.consumers.consumer.profiling;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.util.StopWatch;
+import pl.allegro.tech.hermes.api.SubscriptionName;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * This class is designed to be fully thread safe, except methods <code>startPartialMeasurement</code> and <code>stopPartialMeasurement</code>,
+ * since they are always used in a single thread. Also, method <code>saveRetryDelay</code> is designed to be thread safe,
+ * as <code>retryDelayMillis</code> is modified only by one thread, and it's volatile, so other threads always see updated value.
```

Comment:
does this comments is actual ? every method has `synchronized` and there is no volatile